### ISSUE TITLE
Package control 3.1.0 is cycling on bz2 upgrade

### ISSUE THREAD

**Issue by ilog2000** [wbond/package_control#989]
Hello,

20 minutes ago started Sublime Text 3 (Build 3083) and got a message that Package control upgraded to v3.1.0. There also came a problem though. I constantly get the next sequence of error messages:

1. An error occurred while trying to upgrade bz2. Please restart...
2. The package specified, bz2, is not available.
3. 1 missing dependency was just installed. Sublime Text should be restarted...
and repeating all three steps after restart. On step 1 there is an attempt to clean up "\Packages\bz2" so there only "st3_windows_x64" sub-folder and "dependency-metadata.json" file remains.

Windows 7 64-bit

**Comment by ilog2000** [comment]
Additionally, the "0_package_control_loader.sublime-package" file contains 11 entries of "02-bz2.py"  inside (not sure how it's possible to have 11 entries with the same name in archive, maybe Total Commander archive manager is wrong).

**Comment by ilog2000** [comment]
The issue resolved by deletion of extra 10 ''02-bz2.py" entries from "0_package_control_loader.sublime-package" archive. But it's strange to me how this could happen.

**Comment by wbond** [comment]
My guess is that there was an issue detecting that the bz2 package was installed, although I can't really be sure.

**Comment by FichteFoll** [comment]
>contains 11 entries of "02-bz2.py" inside

I also encountered this a few months ago when messing with dependency installation, but this issue should have since been resolved.

**Comment by wbond** [comment]
If anyone sees an issue like this, please try one of the following based on your OS:

Linux:

```bash
unzip -l ~/.config/sublime-text-3/Installed\ Packages/0_package_control_loader.sublime-package
```

OS X:

```bash
unzip -l ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages/0_package_control_loader.sublime-package
```

Windows:

 - Open Windows Explorer
 - Browse to `%APPDATA%\Sublime Text 3\Installed Packages\`
 - Use 7zip or another archive program to open `0_package_control_loader.sublime-package` to list the files

**Comment by godzig** [comment]
I used wbond's instructions above to open the `0_package_control_loader.sublime-package` archive. Changing the filename from `02-bz2.py` to `01-bz2.py` solved the issue for me.

**Comment by Elwin00** [comment]
I had this problem. I had two files in `Installed Packages` folder on (Windows)
`0_package_control_loader.sublime-package`
`0_package_control_loader.sublime-package-new`
Then I had several `02-bz2.py` inside the first package (the one without -new suffix). The other package had no such files.
Now, after every second application launch, one extra file was added to the package archive.
I deleted all of the `02-bz2.py` files except one but it didn't work, it would keep trying to reinstall bz2.
In the end, I inspected those files and noticed, that the new files are always 68 bytes long, but the original one had 69 bytes. The only difference was a new line at the beginning of the file. So this time I kept the file with first line empty and it seems to work.

**Comment by danieljl** [comment]
Same problem here. I've tried 3 suggestions from @ilog2000, @godzig, @Elwin00 but none seems working for me.

**Comment by wbond** [comment]
@danieljl Can you post the output I asked for with https://github.com/wbond/package_control/issues/989#issuecomment-140375029?

**Comment by chrisburgan** [comment]
Having the same issue. Here's the output from the above command run on OSX. 

unzip -l ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages/0_package_control_loader.sublime-package
Archive:  /Users/username/Library/Application Support/Sublime Text 3/Installed Packages/0_package_control_loader.sublime-package
  Length     Date   Time    Name
 --------    ----   ----    ----
      173  01-06-15 14:36   dependency-metadata.json
     1877  01-06-15 14:36   00-package_control.py
       69  01-06-15 14:36   02-bz2.py
       68  09-16-15 15:52   02-bz2.py
       68  09-16-15 17:25   02-bz2.py
       68  09-16-15 21:17   02-bz2.py
       68  09-17-15 10:37   02-bz2.py
       68  09-17-15 10:41   02-bz2.py
       68  09-17-15 10:42   02-bz2.py
       68  09-17-15 10:42   02-bz2.py
       68  09-17-15 10:44   02-bz2.py
       68  09-17-15 10:47   02-bz2.py
       68  09-17-15 10:49   02-bz2.py
       68  09-17-15 10:53   02-bz2.py
 --------                   -------
     2867                   14 files

**Comment by danieljl** [comment]
@wbond
```
00-package_control.py
02-bz2.py
dependency-metadata.json
```

**Comment by ndi-root** [comment]
I had same issue but it seems solved it in my case. It's just simple though.
I just closed ST3 then deleted
<code> 0_package_control_loader.sublime-package</code>
<code> 0_package_control_loader.sublime-package-new</code>
from Installed Packages folder.
Also I deleted bz2 folder from Packages folder.Then started ST3.
First time I got an attention to restart. So I restart it then any errors didn't show up anymore.

**Comment by danieljl** [comment]
@ndi-root Thanks! It worked for me too.

**Comment by iodic** [comment]
:+1: for the solution provided by @ndi-root, worked for me on OSX. Thanks! :)

**Comment by aMarCruz** [comment]
@ndi-root thanks, worked for me on Win7, 32bits (I need the normal restart only)

**Comment by OrKoN** [comment]
@ndi-root thanks. Worked for me on Linux.

**Comment by nodesocket** [comment]
@ndi-root awesome that worked. For others, here are the commands to fix on OS X.

````
cd ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages
rm -f rm 0_package_control_loader.sublime-package 0_package_control_loader.sublime-package-new
cd ../
cd Packages
rm -rf bz2
```

**Comment by mafredri** [comment]
Thanks @ndi-root! That fixed it, although I only needed to delete the `0_package_control_loader.sublime-package`-files

**Comment by ghprod** [comment]
@ndi-root  Thanks its works for me on Freya 32 bit :dancer:

**Comment by phackwer** [comment]
I removed Package Control, closed Sublime 3, then removed the files and folders indicated @ndi-root, opened Sublime, reinstalled Package Control, restarted Sublime, and only after all that, it worked.

Only removing the indicated files didn't worked. Package Control had to be removed before, and reinstalled at the end.

**Comment by emitategh** [comment]
It Worked por me! Thank you sir!

**Comment by egeerardyn** [comment]
@ndi-root Thanks!

**Comment by bwangel23** [comment]
@wbond ， I solove this questions! Thanks!

**Comment by mitoop** [comment]
The same situation as @ndi-root says , but it doesn't work for me , @godzig it works for me well.

**Comment by aborigeno** [comment]
Also for me the @ndi-root solution didn't work, I've changed the file name inside the zipfile to 01-bz2.py as suggested by @godzig and worked
I'm on OS X 10.10

**Comment by Meekohi** [comment]
@nodesocket works for me. Thanks!

**Comment by beidel** [comment]
@ndi-root worked for me (Windows 7). Thanks!

**Comment by frozenpandaman** [comment]
@ndi-root Also getting this error. However, my "Installed Packages" folder is empty. In Packages/0_package_control_folder, I only have the following files:

00-package_control.py
00-package_control.pyc
02-bz2.py
02-bz2.pyc
50-ordereddict.py
50-ordereddict.pyc
50-requests.py
50-requests.pyc
dependency-metadata.json

I've tried removing this entire folder (0_package_control_folder) along with the Packages/bz2/ folder – they are both re-generated on the first restart of Sublime Text 2, and on subsequent launchings I'm still getting "1 missing dependency was just installed."

**Comment by DoubleU23** [comment]
i re-installed the package control in the sublime console...
fixed it for me (other suggestions didn't work for me)

**Comment by travzdevil69** [comment]
@ndi-root Worked for me on OSX, thanks mate.

**Comment by frozenpandaman** [comment]
Neither @ndi-root's nor @DoubleU23's suggestions are working for me, unfortunately.

**Comment by megadr01d** [comment]
@ndi-root :+1:  (OS X 10.9.5)

**Comment by frozenpandaman** [comment]
This is still happening for me. Below is my console output:

``
Package Control: Installing 1 missing dependency
Package Control: The dependency "fileio" is either not available on this platform or for this version of Sublime Text
Package Control: Installed missing dependency fileio
Package Control: Skipping automatic upgrade, last run at 2015-12-08 10:55:48, next run at 2015-12-08 11:55:48 or after
error: Package Control

1 missing dependency was just installed. Sublime Text should be restarted, otherwise one or more of the installed packages may not function properly.
``

**Comment by frozenpandaman** [comment]
FIXED! Had to add a file named ``01-fileio.py`` in my /Packages/0_package_control_loader/ directory with the following contents:
```
from package_control import sys_path
sys_path.add_dependency(u'fileio')

```

I then created a new directory in /Packages/ named fileio and populated it with the contents of [this](https://github.com/codexns/sublime-fileio) repository.

**Comment by ztyankov** [comment]
@ndi-root Your solution worked for me too. I'm running ST3 on Win 7 64 bit.  Thank you!

**Comment by igregson** [comment]
@Elwin00 Had the same exact problem as you. Deleted `0_package_control_loader.sublime-package-new` and sublime works again.

ST3 on Linux, running Ubuntu 15.10

**Comment by phuna** [comment]
I tried to follow @ndi-root intructions but still same problem.
I did a fresh install. There is no bz2 package in Packages directory. After removing `0_package_control_loader.sublime-package` from Installed Packages and restart ST3, it still says: There are no packages available for installation.

@wbond : your command's result:
  Length     Date   Time    Name
 --------    ----   ----    ----
      173  01-03-16 06:18   dependency-metadata.json
     1876  01-03-16 06:18   00-package_control.py
 --------                   -------
     2049                   2 files

**Comment by vascomgcorreia** [comment]
same problem here.
attempting to install package control, and bz2 dependency not installed.
same output as @phuna

**Comment by mcpengelly** [comment]
Getting the same issue on both OSX and Linux. Should note that is was working fine without issue on both systems for a while until now.

Package control hangs when I try to install new packages or immediately exits and displays:

> Package Control
> There are no packages available for installation

I also get this when i reinstall package control:

> reloading Packages/Package Control/Package Control.sublime-settings
> Package Control: Settings change detected, clearing cache
> Package Control: Installing 1 missing dependency
> Package Control: The dependency specified, bz2, is not available
> Package Control: Skipping automatic upgrade, last run at 2016-01-02 19:19:33, next run at 2016-01- 02 20:19:33

I've reinstalled package control, reinstalled sublime text 3, and trying sublime text 2 also didn't work. The certificates when i check the lock on packagecontrol.io are green. A friend assures me that they are able to accesss the repositories, another has the same issue as me.

**Comment by outliercody** [comment]
Same issues and errors.
I am getting the same exact reults as @phuna 

Tried purging old installs, removing sublime text 3 entirely, reinstalling package control + a combination of all 3 to no avail. Scoured almost all relevant google links as well.

**Comment by nslogan** [comment]
I'm also having this problem. Mac OS X 10.10.5, Sublime Text 3 Build 3095. This surfaced for me while trying to install a package for the first time in a few weeks on my Mac - I got the "There are no packages available for installation" error. I tried reinstalling Package Control, multiple times, to no avail. I then reinstalled Sublime Text 3 entirely, then Package Control. Still no luck. Here's my debug output:

```
reloading plugin 0_package_control_loader.00-package_control
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
plugins loaded
Package Control: Error submitting usage information. URL error EOF occurred in violation of protocol (_ssl.c:550) downloading https://packagecontrol.io/submit?sublime_platform=osx&package_control_version=3.1.2&sublime_version=3095&package=Package+Control&operation=install&version=3.1.2.
Package Control: Installing 1 missing dependency
Package Control: The dependency specified, bz2, is not available
Package Control: Skipping automatic upgrade, last run at 2016-01-03 00:20:25, next run at 2016-01-03 01:20:25 or after
```

I am also able to see the green certificate for packagecontrol.io.

**Comment by nurdeen** [comment]
Same problem as nslogan

**Comment by runithome** [comment]
Same problem as nslogan.

**Comment by adonespitogo** [comment]
Same problem as @nslogan , but on ubuntu 14.04. The problem is there is no bz2 folder under Packages directory, and no 02_bz2.py in `0_package_control_loader.sublime-package`.

**Comment by mcpengelly** [comment]
if this is happening to you within the last two days. Here's why.

Will Bond ‏@wbond  13h13 hours ago
Just kidding, we are down again. Great time to mention I'm looking to sign up supporters at $2.50/month for more distributed infrastructure.

Package control servers are down.

**Comment by adonespitogo** [comment]
@mcpengelly When can the servers be up again?

**Comment by mcpengelly** [comment]
see:
http://status.linode.com/

**Comment by BenMann** [comment]
Had the same issue, seems resolved right now! :dancer:

**Comment by zephyrzoom** [comment]
@ndi-root work for me. Thanks.

**Comment by phuna** [comment]
problem seems resolved now :+1:

**Comment by runithome** [comment]
issue resolved for me as well.

**Comment by AnthonyMichaelc** [comment]
Deleting the 0_package_control_loader.sublime-package files from the AppData\Roaming\Sublime Text 3\Installed Packages and then opening ST3 once to load the files and then reload it to finish the update solved my error issues. This solution also seems to have stopped my browser sync package from not running simple javascript click events.

**Comment by bcrotty** [comment]
The same issue is happening for me right now. It just started 10 min ago.

**Comment by AnthonyMichaelc** [comment]
Did you attempt the steps I listed? What have you tried to resolve this issue yourself, so we can filter out pointless solutions

**Comment by bcrotty** [comment]
I reinstalled Sublime and all packages from scratch, and the issue went away. I'll let you know if it comes back. Thanks!

**Comment by Glutanimate** [comment]
@AnthonyMichaelc Thanks for writing this. Deleting the `0_package_control_loader.sublime-package` files worked for me.

**Comment by AnthonyMichaelc** [comment]
@Glutanimate No problem, It was a simple solution to an annoying problem I encountered. If you do the latest update you shouldn't be prompted to with the same issue again but if so this technique should work agian.

**Comment by ctf0** [comment]
just had the same thing today for no logical reason, removing `0_package_control_loader.sublime-package` did the trick, thanx

**Comment by AnthonyMichaelc** [comment]
@ctf0 No problem it is a simple fix but should stop it from happening again. I have not any of these issues since I did this.

**Comment by Marketionist** [comment]
Just got the same problem on **Mac OS X**, Sublime Text 3, Build 3103.

**The solution**
Delete `0_package_control_loader.sublime-package`, `0_package_control_loader.sublime-package-new` files from Installed Packages folder. Then delete `bz2` folder from Packages folder. Here is how to do it with one terminal command:

`rm -rf ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages/0_package_control_loader.sublime-package ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages/0_package_control_loader.sublime-package-new ~/Library/Application\ Support/Sublime\ Text\ 3/Packages/bz2/`

Then quit Sublime Text and open it again.

**Comment by diegolamanno** [comment]
@ndi-root it worked for me too. Awesome!

**Comment by kelwenzera** [comment]
Just giving my feedback.
Same error message here as the first post said ( me @ Windows 8.1 ).

Steps/What I did:

1 - Opened Sublime, uninstalled all plugins (Cut the plugins).
Restarted Sublime. Same error. Closed it.

2 - Opened it again and installed all plugins again (Paste the plugins and closed Sublime right away, didn't let the plugins actually install).

3 - Went to windows explorer, **%APPDATA%\Sublime Text 3\Installed Packages\**
It had **only two** files named "0_package_control_loader.sublime-package" but one had the "-new" at the end. Deleted the "-new" since it was the older timestamped.

4 - Opened Sublime, error is gone, plugins started to get installed again, everything seems to be working fine.

**I don't know which steps was really necessary**, but I think if you uninstall the plugins, delete the file, and the open it again, it will work.

**Comment by trickbz** [comment]
kelwenzera, just the #3 step helped me. 10x.

**Comment by xs47968224** [comment]
I had same issue but it seems solved it in my case.
I just used 7zip to unzip "0_package_control_loader.sublime-package" file
 @wbond ,thanks, It's helpful

**Comment by dlsso** [comment]
@ndi-root Thanks, fixed based on your solution.

For others on OSX:
All I needed to do was remove ` 0_package_control_loader.sublime-package` and `0_package_control_loader.sublime-package-new` from `~/Library/Application Support/Sublime Text 3/Installed Packages`

**Comment by zgoda** [comment]
I get this message then plugin host crashes (linux 64 bit).

**Comment by bramchi** [comment]
I did have a similar issue with the backrefs package. Manually deleting it from the Packages directory helped here too, the message showed one more time on the next start, and after a Sublime restart it was gone. Sweet! Thanks for the pointers, y'all!

**Comment by burdiyan** [comment]
I had similar issues with backrefs and bz2. Removing these packages solved the issues the same way @bramchi is saying.

**Comment by heyjohnsexton** [comment]
@ndi-root worked for me!

**Comment by omargourari** [comment]
@ndi-root worked for me too
removed the files
started sublime
same error message prompt
restarted
no additional notification

**Comment by FichteFoll** [comment]
Today, this issue occured to me again. I backed up my `0_package_control_loader.sublime-package` and  `0_package_control_loader.sublime-package-new` files so that I will able to reproduce the problem and start investigating in a fix.

**Comment by wbond** [comment]
The most recent version of Package Control no longer requires the bz2 dependency. Additionally I've solved some issues with updating the `0_package_control_loader.sublime-package` file.

Hopefully between the two of these the next version of PC should solve this issue.

**Comment by wbond** [comment]
Here is the commit that should hopefully fix the issue: 0cd0b47f5c2254bf969081edf52bdaba834d04b6

**Comment by wbond** [comment]
This commit may resolve the issue for some other situations: e8b1001bc9cf403312e83a074f330fbd6623a557

**Comment by wbond** [comment]
This should be resolved by 3.2.0. If you are still seeing issues after upgrading and restarting Sublime Text, please:

 1. Open a new issue
 2. Include a full debug log

**Comment by linuodoc** [comment]
Package Control: Installing 1 missing dependency
Package Control: The dependency "paramiko" is either not available on this platform or for this version of Sublime Text
Package Control: Installed missing dependency paramiko
error: Package Control
1 missing dependency was just installed. Sublime Text should be restarted, otherwise one or more of the installed packages may not function properly.
Package Control: No updated packages

this attached annoying error keeps popup every time when launching subtext 3 
Please support?
![Capture](https://user-images.githubusercontent.com/20270953/54139386-dd1d7d00-443a-11e9-99cd-59debcbda7e0.PNG)

**Comment by AbhimanyuAryan** [comment]
deleting the package fixed the issue. Sublime will redownload the package on fresh launch
